src/compiler/sys.ts

```diff
@@ -138,96 +531,104 @@ namespace ts {
-                }
-
-                function fileEventHandler(eventName: string, relativeFileName: string | undefined, baseDirPath: string) {
-                    // When files are deleted from disk, the triggered "rename" event would have a relativefileName of "undefined"
-                    const fileName = !isString(relativeFileName)
-                        ? undefined
-                        : getNormalizedAbsolutePath(relativeFileName, baseDirPath);
-                    // Some applications save a working file via rename operations
-                    if ((eventName === "change" || eventName === "rename")) {
-                        const callbacks = fileWatcherCallbacks.get(fileName);
-                        if (callbacks) {
-                            for (const fileCallback of callbacks) {
-                                fileCallback(fileName, FileWatcherEventKind.Changed);
-                            }
+                    catch (e) {
+                        return undefined;
+                    }
+                },
+                createHash: _crypto ? createMD5HashUsingNativeCrypto : generateDjb2Hash,
+                getMemoryUsage() {
+                    if (global.gc) {
+                        global.gc();
+                    }
+                    return process.memoryUsage().heapUsed;
+                },
+                getFileSize(path) {
+                    try {
+                        const stat = _fs.statSync(path);
+                        if (stat.isFile()) {
+                            return stat.size;
                         }
                     }
+                    catch { /*ignore*/ }
+                    return 0;
+                },
+                exit(exitCode?: number): void {
+                    process.exit(exitCode);
+                },
+                realpath(path: string): string {
+                    try {
+                        return _fs.realpathSync(path);
+                    }
+                    catch {
+                        return path;
+                    }
+                },
+                debugMode: some(<string[]>process.execArgv, arg => /^--(inspect|debug)(-brk)?(=\d+)?$/i.test(arg)),
+                tryEnableSourceMapsForHost() {
+                    try {
+                        require("source-map-support").install();
+                    }
+                    catch {
+                        // Could not enable source maps.
+                    }
+                },
+                setTimeout,
+                clearTimeout,
+                clearScreen: () => {
+                    process.stdout.write("\x1Bc");
                 }
-            }
-            const watchedFileSet = createWatchedFileSet();
+            };
 
-            const nodeVersion = getNodeMajorVersion();
-            const isNode4OrLater = nodeVersion >= 4;
+            nodeSystem.watchFile = getWatchFile();

```

Comment:
do you need nodeSystem in these function calls? can we just use the functions directelly and  move these in the object literal above?